Sources/Basic/Path.swift

```diff
@@ -433,6 +433,20 @@ struct PathImpl {
     }
 }
 
+/// Describes a way in which a path is invalid.
+public enum PathError: Error {
+    case startsWithTilda(path: String)
+}
+
+extension PathError: CustomStringConvertible {
+    public var description: String {
+        switch self {
+        case .startsWithTilda(let path):
+            return "path '\(path)' is not supported; can not start with '~'."
```

Comment:
How about: `invalid path '\(path)'; absolute path must begin with /`

Reply:
Makes sense especially as it is validation for `AbsolutePath`